internal/services/applications/migrations/application_identifier_uri_resource.go

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package migrations
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-provider-azuread/internal/helpers/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azuread/internal/services/applications/parse"
+)
+
+type ResourceApplicationIdentifierUriStateUpgradeV0 struct{}
+
+func (ResourceApplicationIdentifierUriStateUpgradeV0) Schema() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"application_object_id": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+		},
+
+		"key_id": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+			Computed: true,
+			ForceNew: true,
+		},
+
+		"description": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+			Computed: true,
+			ForceNew: true,
+		},
+
+		"value": {
+			Type:      pluginsdk.TypeString,
+			Required:  true,
+			ForceNew:  true,
+			Sensitive: true,
+		},
+
+		"start_date": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+			Computed: true,
+			ForceNew: true,
+		},
+
+		"end_date": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			Computed:     true,
+			ForceNew:     true,
+			ExactlyOneOf: []string{"end_date_relative"},
+		},
+
+		"end_date_relative": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ForceNew:     true,
+			ExactlyOneOf: []string{"end_date"},
+		},
```

Comment:
Doesn't this have to match the schema of the resource exactly? When I look at the resource schema I only see two properties https://github.com/hashicorp/terraform-provider-azuread/blob/8cead2417164648326d3ba16ab8f1cce8a6b4080/internal/services/applications/application_identifier_uri_resource.go#L55-L71
Unless I'm missing something? 🙈

Reply:
@manicminer - This migration schema looks like it belongs to `azuread_application_certificate`?